torchtune/utils/checkpoint.py

```diff
@@ -22,6 +22,59 @@
 )
 
 
+class OptimizerInBackwardWrapper:
```

Comment:
I'd move this to _checkpointer/_checkpoint_utils.py since I plan to eventually nuke this file